homeassistant/components/nanoleaf/config_flow.py

```diff
@@ -0,0 +1,227 @@
+"""Config flow for Nanoleaf integration."""
+from __future__ import annotations
+
+import logging
+import os
+from typing import Any, Final, cast
+
+from pynanoleaf import InvalidToken, Nanoleaf, NotAuthorizingNewTokens, Unavailable
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import persistent_notification
+from homeassistant.const import CONF_HOST, CONF_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.typing import DiscoveryInfoType
+from homeassistant.util.json import load_json, save_json
+
+from .const import DOMAIN
+from .util import pynanoleaf_get_info
+
+_LOGGER = logging.getLogger(__name__)
+
+# For discovery integration import
+CONFIG_FILE: Final = ".nanoleaf.conf"
+
+USER_SCHEMA: Final = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Nanoleaf config flow."""
+
+    VERSION = 1
+
+    nanoleaf: Nanoleaf
```

Comment:
These should probably be defined in __init__ as I don't think we want these to be class variables here

Reply:
Changed in https://github.com/home-assistant/core/pull/52199/commits/64609cd2f1844ebddf1e611e11ce04b6c7f34334.